extensions/itunes/src/album.tsx

```diff
@@ -0,0 +1,253 @@
+import { AbortError } from "node-fetch";
+import fs from "fs";
+import readline from "readline";
+import { execSync } from "child_process";
+import os from "os";
+
+interface Preferences {
+  path: string;
+}
+
+interface SearchState {
+  results: SearchResult[];
+  isLoading: boolean;
+}
+
+interface SearchResult {
+  id: string;
+  artist: string;
+  album: string;
+}
+
+interface runScriptProps {
+  album: string;
+  artist: string;
+}
+
+interface searchEntry {
+  artist: string;
+  album: string;
+}
+
+let match: searchEntry[] = [];
+const preferences: Preferences = getPreferenceValues();
+const using = "iTunes";
+const XMLPath = preferences.path.replace(/^\s/, "").replace(/^~/, os.homedir());
+if (!fs.existsSync(XMLPath)) {
+  showToast(
+    ToastStyle.Failure,
+    "Library XML File Not Found",
+    "Please correct the XML file path in extension preferences."
+  ).then();
+  throw "Invalid XML file path";
+}
+
+const RunScript: FC<runScriptProps> = (props) => {
+  useEffect(() => {
+    const escapingNames = {
+      album: props.album.replace(/"/g, `\\"`),
+      artist: props.artist.replace(/"/g, `\\"`),
+    };
+    try {
+      execSync(`osascript \
+     -e "tell application \\"${using}\\"" \
+     -e "if (exists playlist \\"ExtensionAlbumPlaying\\") then" \
+     -e "delete playlist \\"ExtensionAlbumPlaying\\"" \
+     -e "end if" \
+     -e "set name of (make new playlist) to \\"ExtensionAlbumPlaying\\"" \
+     -e "set theseTracks to every track of library playlist 1 whose album is \\"${escapingNames.album}\\" and artist is \\"${escapingNames.artist}\\"" \
+     -e "repeat with thisTrack in theseTracks" \
+     -e "duplicate thisTrack to playlist \\"ExtensionAlbumPlaying\\"" \
+     -e "end repeat" \
+     -e "play playlist \\"ExtensionAlbumPlaying\\"" \
+     -e "end tell"`);
+      popToRoot().then();
+      closeMainWindow({ clearRootSearch: true }).then();
+    } catch (error) {
+      console.error("execution error", error);
+      showToast(
+        ToastStyle.Failure,
+        "Failed to play the track",
+        "Maybe the song is not in you library or has weird title"
+      ).then();
+      return;
+    }
+  });
+  return <Detail markdown={`# Calling ${using}...`} />;
+};
+
+export default function Command() {
+  const { state, search } = useSearch();
+
+  return (
+    <List
+      isLoading={state.isLoading}
+      onSearchTextChange={search}
+      searchBarPlaceholder="Search by name..."
+      throttle
+    >
+      <List.Section title="Results" subtitle={state.results.length + ""}>
+        {state.results.map((searchResult) => (
+          <SearchListItem key={searchResult.id} searchResult={searchResult} />
+        ))}
+      </List.Section>
+    </List>
+  );
+}
+
+function SearchListItem({ searchResult }: { searchResult: SearchResult }) {
+  return (
+    <List.Item
+      title={searchResult.album}
+      subtitle={searchResult.artist}
+      actions={
+        <ActionPanel>
+          <PushAction
```

Comment:
what's the reason to use a `PushAction` here? Shouldn't it be a regular `ActionPanel.Item` with a callback?